### PR TITLE
Delete hourly limits and add override mechanism for deployment specfic limits

### PR DESCRIPTION
**Pull request by perryibm**
While ansible does not have an override mechanism, this can be accomplished by grouping keys together in WhiskConfig.  PG1-408

### CODE REVIEW

File: core/controller/src/main/scala/whisk/core/entitlement/ActivationThrottler.scala [hunk 1]
```diff
@@ -41,6 +41,8 @@ import whisk.core.entity.Subject
 class ActivationThrottler(consulServer: String, concurrencyLimit: Int)(
     implicit val system: ActorSystem) extends Logging {
 
+    info(this, s"ActivationThrottler.concurrencyLimit = $concurrencyLimit")
```
Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/entitlement/ActivationThrottler.scala]: `this` will print `ActivationThrottler`.

File: core/controller/src/main/scala/whisk/core/entitlement/RateThrottler.scala [hunk 2]
```diff
@@ -28,8 +28,9 @@ import akka.event.Logging.InfoLevel
  *
  * For now, we throttle only at a 1-minute granularity.
  */
-class RateThrottler(maxPerMinute: Int,
-                    maxPerHour: Int) extends Logging {
+class RateThrottler(maxPerMinute: Int) extends Logging {
+
+    info(this, s"RateThrottler.maxPerMinute = $maxPerMinute")
```
Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/entitlement/RateThrottler.scala]: `RateThrottler` not necessary. Will change when merging.

**Comment by perryibm** [comment]
PG1-483 (and 484) show it working.  Note that the throttle test case for concurrent checking now performs exactly N+1 invokes and is thus "tight".

File: tests/src/limits/ThrottleTests.scala [hunk 3]
```diff
@@ -55,13 +55,27 @@ class ThrottleTests
...
     val concurrencyMessage = "The user has sent too many requests in a given amount of time."
 
+    /*
+     * Retrieve a numeric limit for the key from the property set.  If the overrideKey is present, use that.
+     */
+    def getLimit(key: String, overrideKey: String) = try {
+        WhiskProperties.getProperty(overrideKey).toInt
+    } catch {
+        case _: Throwable => WhiskProperties.getProperty(key).toInt
+    }
```
Comment (markusthoemmes) [inline tests/src/limits/ThrottleTests.scala]: I think, using the "functional" way, a `Try` makes this more readable (and idiomatic).

```
def getLimit(key: String, overrideKey: String) = Try {
    WhiskProperties.getProperty(overrideKey).toInt
} getOrElse {
    WhiskProperties.getProperty(key).toInt
}
```

**Comment by markusthoemmes** [comment]
In general: Has it been verified that this override approach is necessary and cannot completely be done in ansible? @lehoanganh @domdom82 ?

**Comment by perryibm** [comment]
Yes, I have talked to both of them about this.

File: tests/src/limits/ThrottleTests.scala (hunk 3 above)
Comment (perryibm) [inline tests/src/limits/ThrottleTests.scala]: Did you mean `...orElse....getOrElse`  since `value` may also fail?

**Comment by perryibm** [comment]
retesting one last time at PG1-487

**Comment by perryibm** [comment]
Last travis build got stuck building swift image. Re-trying.

**Comment by markusthoemmes** [comment]
In a quick chat with @domdom82 and @lehoanganh they stated that this would be possible by removing the global "default" and use a specific limit for each deployment. Would it make sense to do that instead?

**Comment by rabbah** [comment]
We considered this. Since the limits are overridden to a lower value in only a very small number of deployments (1 currently), it makes more sense to do it this way rather than modify `N` environment files which all will have the same value.

File: common/scala/src/main/scala/whisk/common/Config.scala [hunk 4]
```diff
@@ -39,20 +39,26 @@ package whisk.common
...
      * Gets value for key if it exists else the empty string.
+     * The value of the override key will instead be returned if its value is present in the map.
      *
      * @param key to lookup
+     * @param overrideKey the property whose value will be returned if the map contains the override key.
      * @return value for the key or the empty string if the key does not have a value/does not exist
      */
-    def apply(key: String): String = try {
-        settings(key)
+    def apply(key: String, overrideKey: String = ""): String = try {
```
Comment (markusthoemmes) [inline common/scala/src/main/scala/whisk/common/Config.scala]: Is this the same as

`Try(settings(overrideKey)).orElse(Try(settings(key)).getOrElse("")` ?

File: tests/src/limits/ThrottleTests.scala (hunk 3 above)
Comment (perryibm) [inline tests/src/limits/ThrottleTests.scala]: yep - done - i misread it the first time thinking it was for some other file

**Comment by perryibm** [comment]
Travis now passing the final throttle test.  Redoing PG just in case - PG1-501.

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/dbc0e8f997fe67d3624ce9d02de287fd76d53d96.

**Pull request closed without merging**